test-support/src/main/java/org/testcontainers/testsupport/Flaky.java

```diff
@@ -0,0 +1,34 @@
+package org.testcontainers.testsupport;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotation for test methods that should be retried in the event of failure. See {@link FlakyTestJUnit4RetryRule} for
+ * more details.
+ */
+@Retention(RUNTIME)
+@Target({METHOD})
+public @interface Flaky {
+
+    /**
+     * @return a URL for a GitHub issue where this flaky test can be discussed, and where actions to resolve it can be
+     * coordinated.
+     */
+    String githubIssueUrl();
+
+    /**
+     * @return a date at which this should be reviewed, in {@link java.time.format.DateTimeFormatter#ISO_LOCAL_DATE}
+     * format (e.g. {@code 2020-12-03}). Now + 3 months is suggested. Once this date has passed, retries will no longer
+     * be applied.
+     */
+    String reviewDate();
```

Comment:
WDYT about letting them fail explicitly instead of not retrying?
Which would mean, we either have to remove the annotation or prolong it?

Or does this create too much maintenance? I think the maintenance should be the same, just without explicitly failing, we will miss the flaky tests for some time.

Reply:
I suspect it could end up being a PITA for contributors, to be honest!